### PR TITLE
Reader for MNIST files. 

### PR DESCRIPTION
**Pull request by kailamink**
Parses initial compressed file into  output directory in the format MNISTOutputFiles / <Number> / <FileNumber> 

Reader is re-runnable, prints any errors that occur to the log screen, and creates output directories if they don't already exist.

### CODE REVIEW

File: example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package andrewoid.neuralnetwork.MNISTReader;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+public class IdxReader
```
Comment (andrewoid) [inline example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java:11]: If you copied this code from somewhere, include a link to it in a javadoc comment to the class.

File: example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+{
+
+    public static String inputImagePath;
```
Comment (andrewoid) [inline example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java]: These should not be `public` and `static`

File: example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    public static String outputPath;
+
+    public IdxReader(String inputImagePath, String inputLabelPath, String outputPath)
+    {
+        this.inputImagePath = inputImagePath;
+        this.inputLabelPath = inputLabelPath;
+        this.outputPath = outputPath;
+    }
+
+    public void loadFromCompressedFilesToOutputDir() {
```
Comment (andrewoid) [inline example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java:42]: this method is really long, please break it up into smaller methods

File: example/src/main/java/andrewoid/neuralnetwork/CalculatorNetworkTraining.java [hunk 4]
```diff
@@ -50,8 +51,15 @@
             {0, 0, 0, 0, 0, 0, 0, 0, 0, 1}, // 9
     };
 
-    public static void main(String args[]) {
+    public static final String inputImagePath = "train-images-idx3-ubyte";
+    public static final String inputLabelPath = "train-labels-idx1-ubyte";
+    public static final String outputPath = "MNISTOutputFiles/";
 
+    public static void main(String args[])
```
Comment (andrewoid) [inline example/src/main/java/andrewoid/neuralnetwork/CalculatorNetworkTraining.java:54]: Make you own `main()`, don't reuse `CalculatorNetworkTraining.java`

File: example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        }
+    }
+
+    private void createOutputDirectories(String outputPath)
+    {
+        File outputDir = new File(outputPath);
+        outputDir.mkdir();
+        for(int ix = 0; ix < 10; ++ix)
+        {
+            new File(outputPath + "" + ix).mkdir();
```
Comment (andrewoid) [inline example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java:130]: `+ ""` isn't necessary here

Comment (kailamink) [reply example/src/main/java/andrewoid/neuralnetwork/MNISTReader/IdxReader.java:130]: It wasn't allowing me to add them otherwise

**Review by andrewoid — APPROVED** [review]

**Pull request merged**